toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -0,0 +1,103 @@
+import os
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import requests
+from pydantic import Field
+
+from toucan_connectors.oauth2_connector.oauth2connector import (
+    OAuth2Connector,
+    OAuth2ConnectorConfig,
+)
+from toucan_connectors.toucan_connector import (
+    ConnectorSecretsForm,
+    ToucanConnector,
+    ToucanDataSource,
+)
+
+API_BASE_ROUTE: str = 'https://graph.microsoft.com/v1.0/me'
+AUTHORIZATION_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize'
+TOKEN_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/token'
+SCOPE: str = 'offline_access Files.Read'
+
+
+class OneDriveDataSource(ToucanDataSource):
+    file: str
+    sheet: str
+    range: Optional[str]
+
+
+class OneDriveConnector(ToucanConnector):
+
+    data_source_model: OneDriveDataSource
+
+    _auth_flow = 'oauth2'
+    _oauth_trigger = 'instance'
+    oauth2_version = Field('1', **{'ui.hidden': True})
+    auth_flow_id: Optional[str]
+
+    def __init__(self, **kwargs):
+        super().__init__(
+            **{k: v for k, v in kwargs.items() if k not in OAuth2Connector.init_params}
+        )
+        # we use __dict__ so that pydantic does not complain about the _oauth2_connector field
+        self.__dict__['_oauth2_connector'] = OAuth2Connector(
+            auth_flow_id=self.auth_flow_id,
+            authorization_url=AUTHORIZATION_URL,
+            scope=SCOPE,
```

Comment:
So if scope is configurable, here we should use the value configured for the connector. 
`scope=self.scope`